admin/server/reports.go

```diff
@@ -787,6 +822,20 @@ func parseReportAnnotations(annotations map[string]string) reportAnnotations {
 	res.AdminNonce = annotations["admin_nonce"]
 	res.WebOpenPath = annotations["web_open_path"]
 	res.WebOpenState = annotations["web_open_state"]
+	res.Explore = annotations["explore"]
+	res.Canvas = annotations["canvas"]
+	switch annotations["web_open_mode"] {
+	case "recipient":
+		res.WebOpenMode = WebOpenModeRecipient
+	case "creator":
+		res.WebOpenMode = WebOpenModeCreator
+	case "none":
+		res.WebOpenMode = WebOpenModeNone
+	case "filtered":
+		res.WebOpenMode = WebOpenModeFiltered
+	case "": // backwards compatibility
+		res.WebOpenMode = WebOpenModeRecipient
```

Comment:
See comment above – since you have the `WebOpenMode` enum type, ideally you can just cast to that, e.g. something like:
```go
res.WebOpenMode := WebOpenMode(annotations["web_open_mode"])
if res.WebOpenMode == "" {
  res.WebOpenMode = WebOpenModeRecipient // Backwards compatibility
}
if !res.WebOpenMode.Valid() {
  return fmt.Errorf("...")
}
```